### PR TITLE
9402 advanced search filter

### PR DESCRIPTION
**Pull request by cooperka**
Integrates the existing search box into the new Filters component.

### CODE REVIEW

File: spec/javascript/components/search/__snapshots__/AdvancedSearchFilter.test.js.snap [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+<div>
+  <input
+    autoComplete="off"
+    className="form-control"
+    id="search_str"
+    name="search"
+    onChange={[MockFunction]}
+    placeholder="Advanced search"
+    type="text"
+    value="query"
```
Comment (smoyth) [inline spec/javascript/components/search/__snapshots__/AdvancedSearchFilter.test.js.snap]: Shouldn't this be `foo`? I'm not seeing `foo` anywhere in this snapshot even though that's the query param...

File: app/javascript/components/search/Filters.js [hunk 2]
```diff
@@ -38,6 +47,14 @@ class Filters extends React.Component {
     this.setState({selectedFormIds: []});
   }
 
+  handleChangeAdvancedSearch(event) {
+    this.setState({advancedSearchText: event.target.value});
+  }
+
+  handleClearFilters() {
+    submitSearch(null);
```
Comment (smoyth) [inline app/javascript/components/search/Filters.js]: When clearing search, can we remove the search query string param (and even the `?` if it's the only param) please? I'm sure there is an npm package that will do that for you!

File: spec/javascript/components/search/__snapshots__/AdvancedSearchFilter.test.js.snap (hunk 1 above)
Comment (smoyth) [reply spec/javascript/components/search/__snapshots__/AdvancedSearchFilter.test.js.snap]: OH RIGHT. I guess this snapshot will change once we integrate with the backend.

**Comment by smoyth** [comment]
@cooperka Hmm, it seems there are a lot of CI failures. We will need to figure those out...

File: spec/javascript/components/search/__snapshots__/AdvancedSearchFilter.test.js.snap (hunk 1 above)
Comment (cooperka) [reply spec/javascript/components/search/__snapshots__/AdvancedSearchFilter.test.js.snap]: Actually this value is coming from the test mock in `spec/.../search/utils.js` which defines `advancedSearchProps`. The AdvancedSearchFilter component itself gets props directly, it doesn't care what the actual url querystring is.

**Comment by cooperka** [comment]
@smoyth tests should finally be passing. There were a lot of weird inconsistencies with the way Ruby talks to React, and also with the way Capybara manipulates the DOM.

I squashed the hell out of the commits, hopefully the solution is fairly readable now. Ready for review.

**Pull request merged**